neurons/validator.py

```diff
@@ -632,8 +633,15 @@ async def run_step(self):
             7. Logs all relevant data for the step, including model IDs, pages, batches, wins, win rates, and losses.
         """
 
-        competition = constants.COMPETITION_SCHEDULE[
-            self.global_step % len(constants.COMPETITION_SCHEDULE)
+        # Note that block on the metagraph only updates on sync operations.
```

Comment:
For future we could consider a sync_now() method on the syncer that we call before a run_step.

Reply:
I considered something like that, although it could add some delay in the regular case to sometimes help adjust a competition schedule check up by 20 minutes.

Could theoretically have logic that also checks if the current block is within x blocks of a future competition though to trigger it. Will add a note to our todos about this at least.